common/src/chain/block/block_v1.rs

```diff
@@ -4,17 +4,15 @@ use crate::chain::transaction::Transaction;
 use crate::chain::ChainConfig;
 use crate::primitives::id;
 use crate::primitives::id::Idable;
-use crate::primitives::{Id, H256};
+use crate::primitives::{Id, Version, H256};
 
 use serialization::{Decode, Encode};
 
 use super::consensus_data::BlockRewardTransactable;
-use super::BlockVersion;
 
-#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd, Encode, Decode)]
+#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd, Encode, Decode, serialization::Tagged)]
 pub struct BlockHeader {
-    #[codec(compact)]
-    pub(super) block_version: u32,
+    pub(super) version: Version<1>,
```

Comment:
If I understand correctly how this works, the value of the version for this block is 0x01 (maybe we should choose 0x00), attempting to deserialize data that leads to some other value, say 0x02, in the version will cause a deserialization error.

Reply:
Correct. Serializing `Version<1>` generates one byte of 0x01, deserializing consumes 0x01 and fails if it's not there

Reply:
I chose 1 for compatibility. In block it does not matter because there are more changes for block. In Transaction, it means the encoding is same as it was before

Reply:
Let's change it some time later.